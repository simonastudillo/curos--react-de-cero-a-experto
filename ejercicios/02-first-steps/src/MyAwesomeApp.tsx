import type { CSSProperties } from "react";

const firstName = 'Fernando';
const lastName = 'Herrera';

const favoriteGames = ['Elden Ring', 'Smash', 'Metal Gear'];
const isActive = true;

const address = {
   zipCode: 'ABC-123',
   country: 'CanadÃ¡'
}
const myStyles: CSSProperties = {
   backgroundColor: '#ff00ff',
   borderRadius: isActive ? 20 : 10,
   padding: 10,
   marginTop: 30
}

export const MyAwesomeApp = () => {

   return (
      <div data-testid="div-app">
         <h1 data-testid="first-name-title"> {firstName} </h1>
         <h3 data-testid="last-name-title"> {lastName} </h3>

         <p>{2 + 2}</p>
         <p className="mi-clase-favorita">{favoriteGames.join(', ')}</p>

         <h1>{isActive ? 'Activo' : 'No activo'}</h1>
         <p
            style={myStyles}
         >{JSON.stringify(address)}</p>
      </div>
   );
}