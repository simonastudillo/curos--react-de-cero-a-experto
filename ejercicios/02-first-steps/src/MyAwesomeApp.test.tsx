import { render, screen } from "@testing-library/react";
import { describe, test, expect } from 'vitest';
import { MyAwesomeApp } from "./MyAwesomeApp";


describe('MyAwesomeApp', () => {

   test('should render firstName and lastName', () => {
      // Arrange
      const expectFirstName = 'Fernando';
      const expectLastName = 'Herrera';
      // Act
      const { container } = render(<MyAwesomeApp />);
      const h1 = container.querySelector('h1');
      const h3 = container.querySelector('h3');
      // Assert
      expect(h1?.innerHTML).toContain(expectFirstName);
      expect(h3?.innerHTML).toContain(expectLastName);
   })

   test('should render firstName and lastName - screen', () => {
      // Arrange
      const expectFirstName = 'Fernando';
      const expectLastName = 'Herrera';
      // Act
      render(<MyAwesomeApp />);
      const h1 = screen.getByTestId('first-name-title');
      const h3 = screen.getByTestId('last-name-title');
      // Assert
      expect(h1?.innerHTML).toContain(expectFirstName);
      expect(h3?.innerHTML).toContain(expectLastName);
   })

   test('should match snapshot', () => {
      // Arrange
      const { container } = render(<MyAwesomeApp />);
      // Assert
      expect(container).toMatchSnapshot();
   })

   test('should match snapshot - screen', () => {
      // Arrange
      render(<MyAwesomeApp />);
      // Act
      const div = screen.getByTestId('div-app');
      // Assert
      expect(div).toMatchSnapshot();
   })

});