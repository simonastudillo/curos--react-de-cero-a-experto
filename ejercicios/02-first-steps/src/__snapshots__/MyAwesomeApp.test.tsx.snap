// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Fernando
     
  </h1>
  <h3
    data-testid="last-name-title"
  >
     
    Herrera
     
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Metal Gear
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(255, 0, 255); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Canadá"}
  </p>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      Fernando
       
    </h1>
    <h3
      data-testid="last-name-title"
    >
       
      Herrera
       
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gear
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: rgb(255, 0, 255); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Canadá"}
    </p>
  </div>
</div>
`;
